### DISCUSSION TITLE
Debounce question

### DISCUSSION THREAD

**Discussion by cro** [django-commons/django-unicorn#696] [Q&A]
I have a search component

```
<div>
<input type="text" unicorn:model.debounce-1000.lazy="search" id="search" />
</div>
```

and in the python file

```
    def updated_search(self, search):
        """
        Called after debouncing whenever the search field changes
        """
        self.query = search
        if self.query:
            mq = self._page_query(self.query)
        else:
            self.parent.load_table()
        # other query processing, not relevant for this example
        self.parent.movers = <complete query here>
        self.parent.force_render = True
```

Placing a breakpoint in PyCharm on the update_search function shows it being called for every keypress.  I expected it to be called once (as long as there is less than 1 second between each keypress).  What am I missing here?

**Comment by adamghill** [comment]
Yeah, this is a little confusing. If you watch this example video you see the debounce is working as expected and there is only one AJAX call:

https://github.com/adamghill/django-unicorn/assets/317045/d0d3a684-99fe-4eed-af9f-2e0307c364af

However, there are two "actions" queued up and they each call the `updated_` function which I don't think is what you want to happen. Let me think about another way you can solve what you are trying to do -- I'll update this thread in a few.

**Reply by adamghill** [reply]
@cro I created #697 with one approach to this, but added a few other approaches and questions in the description. What are your thoughts? Do you have any preferences or should I just sleep on it and try to figure out what I think is the best way forward?

**Comment by adamghill** [comment]
@cro I added a `resolved` method and it got released in 0.61.0. Thanks for creating this discussion!

**Comment by cro** [comment]
Fantastic!  I will try it out in the next day or so!
